streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -766,12 +855,49 @@ public synchronized ProcessorTopology build(Integer topicGroupId) {
         if (topicGroupId != null) {
             nodeGroup = nodeGroups().get(topicGroupId);
         } else {
-            // when nodeGroup is null, we build the full topology. this is used in some tests.
-            nodeGroup = null;
+            // when topicGroupId is null, we build the full topology minus the global groups
+            final Set<String> globalNodeGroups = globalNodeGroups();
+            final Collection<Set<String>> values = nodeGroups().values();
+            nodeGroup = new HashSet<>();
+            for (Set<String> value : values) {
+                nodeGroup.addAll(value);
+            }
+            nodeGroup.removeAll(globalNodeGroups);
+
+
         }
         return build(nodeGroup);
     }
 
+    /**
+     * Builds the topology for any global state stores
+     * @return ProcessorTopology
+     */
+    public synchronized ProcessorTopology buildGlobalStateTopology() {
```

Comment:
I am still not sure about this design. Actually, the global part shout use a singleton pattern. It is kinda weird that we have two `ProcessorTopology` "types" -- the global one and the regular one.

As the KIP is still under discussion, we might want to think about this once more. Also from an PAPI vs DSL point of view -- I think, we need a better separation between both and not "pollute" PAPI with DSL concepts.